examples/field_examples/viking_graben_line12/fwi/fwi.jl

```diff
@@ -0,0 +1,278 @@
+# Pkg.add("LinearAlgebra")
+# Pkg.add("Interpolations")
+# Pkg.add("DelimitedFiles")
+# Pkg.add("Distributed")
+# Pkg.add("DSP")
+# Pkg.add("SlimOptim")
+# Pkg.add("NLopt")
+# Pkg.add("HDF5")
+# Pkg.add("SegyIO")
+# Pkg.add("Plots")
+# Pkg.add("SetIntersectionProjection")
+
+using Statistics, Random, LinearAlgebra, Interpolations, DelimitedFiles, Distributed, DSP
+using JUDI, SlimOptim, NLopt, HDF5, SegyIO, Plots
+using SetIntersectionProjection
+
+
+############# INITIAL DATA #############
+modeling_type = "acoustic"    # scalar, acoustic
+frq = 0.005 # kHz
+trc_len = 4000  # ms
+
+prestk_dir = "$(@__DIR__)/filt/$(frq)hz_$(trc_len)ms/"
+dir_out = "$(@__DIR__)/fwi_$(modeling_type)/$(frq)Hz/"
+
+prestk_file = "shot"
+model_file = "$(@__DIR__)/initial_model/model.h5"
+# after each iteration you should set an appropriate model file computed with previous step
+# model_file = "$(@__DIR__)/fwi_$(modeling_type)/0.005Hz/model.h5"
+model_file_out = "model"
+
+# use original wavelet file 
+wavelet_file = "$(@__DIR__)/../FarField.dat" # dt=1, skip=25
+# or use deghosted wavelet
+# wavelet_file = "$(@__DIR__)/../proc/FarField_deghosted.dat" # dt=4, skip=0
+wavelet_skip_start = 25 # 25 [lines] for raw source and 0 for deghosted source
+wavelet_dt = 1          # 1 [ms] for raw source and 4 [ms] for deghosted source
+
+segy_depth_key_src = "SourceSurfaceElevation"
+segy_depth_key_rec = "RecGroupElevation"
+
+seabed = 355  # [m]
+
+############# INITIAL PARAMS #############
+# water velocity, km/s
+global vwater = 1.5
+
+# water density, g/cm^3
+global rhowater = 1.02
+
+# JUDI options
+buffer_size = 0f0    # limit model (meters) even if 0 buffer makes reflections from borders that does't hurt much the FWI result
+
+# prepare folder for output data
+mkpath(dir_out)
+
+# Load data and create data vector
+container = segy_scan(prestk_dir, prestk_file, ["SourceX", "SourceY", "GroupX", "GroupY", "RecGroupElevation", "SourceSurfaceElevation", "dt"])
+d_obs = judiVector(container; segy_depth_key = segy_depth_key_rec)
+
+srcx = Float32.(get_header(container, "SourceX")[:,1])
+grpx = Float32.(get_header(container, "GroupX")[:,1])
+min_src_x = minimum(srcx)./1000f0
+max_src_x = maximum(srcx)./1000f0
+min_grp_x = minimum(grpx)./1000f0
+max_grp_x = maximum(grpx)./1000f0
+min_cdp_x = (min_src_x+min_grp_x)/2f0
+max_cdp_x = (max_src_x+max_grp_x)/2f0
+min_x = minimum([min_src_x, min_grp_x])
+max_x = maximum([max_src_x, max_grp_x])
+
+# helpful functions
+rho_from_slowness(m) = 0.23.*(sqrt.(1f0 ./ m).*1000f0).^0.25
+
+# Load starting model (mlog - slowness built with Vs from logs; mvsp - built from VSP)
+fid = h5open(model_file, "r")
+n, d, o = read(fid, "n", "d", "o")
+if haskey(fid, "mvsp")
+    m0 = read(fid, "mvsp")
+else
+    m0 = read(fid, "m")
+end
+close(fid)
+
+n = Tuple(Int64(i) for i in n)
+d = Tuple(Float32(i) for i in d)
+o = Tuple(Float32(i) for i in o)
+if modeling_type == "scalar"
+    model0 = Model(n, d, o, m0)
+elseif modeling_type == "acoustic"
```

Comment:
Not sure about the `scalar/acoutic` name I would probably call it something like `slowness/bulk` maybe